File: SYP.Web/Modules/Email/EmailLogs/EmailLogsColumns.cs
namespace SYP.Email.Columns;

[ColumnsScript("Email.EmailLogs")]
[BasedOnRow(typeof(EmailLogsRow), CheckNames = true)]
public class EmailLogsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight, Width(80)]
    public long Id { get; set; }

    [Width(250)]
    public string EmailSubject { get; set; }

    [Width(200)]
    public string ToAddress { get; set; }

    [Width(100)]
    public int Status { get; set; }

    [Width(300)]
    public string StatusMessage { get; set; }

    [Width(80)]
    public int Duration { get; set; }

    [Width(150)]
    public DateTime InsertDate { get; set; }
}

File: SYP.Web/Modules/ServerTypes/Email/EmailLogsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { EmailLogsRow } from "./EmailLogsRow";
import { EmailLogStatus } from "./EmailLogStatus";

export interface EmailLogsColumns {
    Id: Column<EmailLogsRow>;
    EmailSubject: Column<EmailLogsRow>;
    ToAddress: Column<EmailLogsRow>;
    Status: Column<EmailLogsRow>;
    StatusMessage: Column<EmailLogsRow>;
    Duration: Column<EmailLogsRow>;
    InsertDate: Column<EmailLogsRow>;
}

export class EmailLogsColumns extends ColumnsBase<EmailLogsRow> {
    static readonly columnsKey = 'Email.EmailLogs';
    static readonly Fields = fieldsProxy<EmailLogsColumns>();
}

[EmailLogStatus]; // referenced types